src/lib/cache/subCache.cpp

```diff
@@ -113,22 +134,44 @@ bool EntityInfo::match
   const std::string&  type
 )
 {
-  //
-  // If type non-empty - perfect match is mandatory
-  // If type is empty - always matches
-  //
-  if ((type != "") && (entityType != type) && (entityType != ""))
-  {
-    return false;
-  }
+  bool matchedType = false;
+  bool matchedId   = false;
 
+  // Check id
   if (isPattern)
   {
     // REGEX-comparison this->entityIdPattern VS id
-    return regexec(&entityIdPattern, id.c_str(), 0, NULL, 0) == 0;
+    matchedId =  (regexec(&entityIdPattern, id.c_str(), 0, NULL, 0) == 0);
+  }
+  else if (id == entityId)
+  {
+    matchedId =  true;
+  }
+  else
+  {
+    matchedId = false;
+  }
```

Comment:
This else (setting matchedId to FALSE), is it needed?
matchedId is set to FALSE upon creation.